spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -297,4 +297,13 @@ public IsolationLevel isolationLevel() {
         confParser.stringConf().option(SparkWriteOptions.ISOLATION_LEVEL).parseOptional();
     return isolationLevelName != null ? IsolationLevel.fromName(isolationLevelName) : null;
   }
+
+  public boolean caseSensitive() {
+    return confParser
+        .booleanConf()
+        .sessionConf(SparkSQLProperties.CASE_SENSITIVE)
+        .option(SparkWriteOptions.CASE_SENSITIVE)
```

Comment:
As I noted below, I think it's fair for us to just pull the value from the sessionConf. I don't think being allowed to override this at a per-write level will be useful.